### ISSUE TITLE
Publish on npm

### ISSUE THREAD

**Issue by levacic** [nakupanda/bootstrap3-dialog#256]
I'd love to install this package via npm, and it's currently possible to do so using [GitHub URLs](https://docs.npmjs.com/files/package.json#github-urls) syntax in `package.json` - but it would be awesome if the package was actually published to npm.

There is already a [bootstrap-dialog](https://www.npmjs.com/package/bootstrap-dialog) package published by someone else, which seems to reference this repo, but is actually currently at 1.34.6, so it hasn't been updated in a while.

The [bootstrap3-dialog](https://www.npmjs.com/package/bootstrap3-dialog) name isn't taken, and since that's the name in this repo's `package.json`, perhaps you could claim that one?

For me, the advantage of using npm instead of Bower is that Bower doesn't have a built-in way of locking dependencies, which has caused me actual problems on multiple occasions, and secondly, I just don't want to use so many package managers - I already need Composer for PHP, Bundler for Ruby, and npm for JS, and it's getting pretty crowded.

**Comment by levacic** [comment]
Forgot to add - using GitHub URLs does allow for installing the package via npm, but it locks down the dependency to a specific version, instead of being able to use semver, e.g. `^1.35.1`, so it requires me to manually update my `package.json` once a newer version is available.

**Comment by nakupanda** [comment]
Thank you, the idea makes sense, I will add it to NPM soon.

**Comment by levacic** [comment]
Hi, any updates on this? I'm having a lot of trouble when shrinkwrapping (locking down the version, if you're not familiar with npm) the library installed directly from GitHub, and it would be really great if I could do it properly through npm.

The errors are likely some npm bug, but it only happens with this library, and it's the only one installed directly from GitHub:

```
npm ERR! Problems were encountered
npm ERR! Please correct and try again.
npm ERR! invalid: have bootstrap3-dialog@1.35.1 (expected: github:nakupanda/bootstrap3-dialog#v1.35.1) /home/levacic/project/node_modules/bootstrap3-dialog
npm ERR! extraneous: bootstrap3-dialog@1.35.1 /home/levacic/project/node_modules/bootstrap3-dialog
```

So I'm pretty certain that this isn't working quite as expected.

Thanks in advance, if I can help in any way, let me know, I'd be glad to do it.

**Comment by nakupanda** [comment]
Hi,

I tried to push it to NPM but failed due to an error, I will try it again later.

**Comment by nakupanda** [comment]
Something wrong happened here when adding bootstrap3-dialog to NPM, if you don't mind can you please add it?

**Comment by levacic** [comment]
I've added it here: https://www.npmjs.com/package/bootstrap3-dialog

If you'd like me to add you as a collaborator, just tell me your username on npm and I'll do that so you can have control over the package (and you can feel free to remove me at that point if you wish).

Once you're added as a collaborator, you'll need to configure your CLI client to be able to authenticate as you when publishing new versions:

```sh
$ npm login
```

This will prompt you for your npm username, password and email.

After that, it's a simple matter of running

```sh
npm publish
```

each time you want to publish a new version (before that, you should update the version in `package.json` and tag it via git).

**Comment by nakupanda** [comment]
Thank you for adding the package and the detailed introductions, I will contact you to add me as a collaborator when I need to update a new version.

:)

**Comment by baconface** [comment]
Since the NPM package `bootstrap-dialog` uses your code and appears to not be regularly maintained I bet if you shoot the NPM team an email they will give the package namespace to you or set up a redirect. I maintained a package previously with code that wasn't mine because it wasn't on NPM. About a year later I got a courtesy email from NPM that the author wanted to take it over. I understood and willingly surrendered to package. This would clear up a lot confusion and issues the `bootstrap-dialog` package is causing.
